app/_components/cbio-util.js

```diff
@@ -666,6 +672,47 @@ cbio.util = (function() {
     return def.promise();
   }
 
+
+  function getDecimalExponents(data){
+
+    // Copy the values, rather than operating on references to existing values
+    if (data === null || data.length < 1) {//if data is null or empty, return data
+      return data;
+    }
+    
+    var values = [];
+    _.each(data, function(item) {
+      if (!isNaN(item)) {
+        values.push(Number(item));
+      }
+    });
+
+    // Then sort
+    values.sort(function(a, b) {
+      return a - b;
+    });
+    
+    var minZeros = 0, maxZeros = 0;
+      
+    while (values[0] < 1) {
```

Comment:
You're right. I will use a for loop to make sure min and max values are valid numbers.